Source/Dafny/AST/DafnyAst.cs

```diff
@@ -369,6 +369,11 @@ public class BuiltIns {
       return s.StartsWith("_tuple#");
     }
     public const string TupleTypeCtorNamePrefix = "_#Make";  // the printer wants this name prefix to be uniquely recognizable
+
+    public static string TupleTypeCtorName(int dims) {
```

Comment:
A similar change could be made to replace the calls `someStringVar.StartsWith(BuiltIns.TupleTypeCtorNamePrefix)` by `BuiltIns.IsTupleTypeName(someStringVar)`, and then `TupleTypeCtorNamePrefix` could be made private.